sale_order_lock_stock_picking/models/stock_picking.py

```diff
@@ -0,0 +1,29 @@
+# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models, api, _
+from odoo.exceptions import ValidationError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def lock_validation_error(self):
+        if self.sale_id.state == 'done':
+            raise ValidationError( _(
+                "The related sales order is locked. Unlock to continue."
+                )
+            )
+
+    def button_validate(self):
+        self.lock_validation_error()
+        super(StockPicking, self).button_validate()        
```

Comment:
Missing return (it's impossible to validate a picking):
```suggestion
        return super(StockPicking, self).button_validate()        
```